Source/engine/render/text_render.cpp

```diff
@@ -109,19 +133,17 @@ void LoadFont(GameFontTables size, text_color color)
 		LoadMaskedArt(path, font.get(), 256, 1);
 	}
 
-	uint32_t fontId = (color << 24) | (size << 16);
-	Fonts.insert(make_pair(fontId, move(font)));
-
-	sprintf(path, "fonts\\%i-00.bin", FontSizes[size]);
-	LoadFileInMem(path, FontKerns[size]);
+	return Fonts.insert(make_pair(fontId, move(font))).first->second.get();
 }
 
-void UnloadFont(GameFontTables size, text_color color)
+} // namespace
+
+void UnloadFonts(GameFontTables size, text_color color)
 {
-	uint32_t fontId = (color << 24) | (size << 16);
+	uint32_t fontStyle = (color << 24) | (size << 16);
 
 	for (auto font = Fonts.begin(); font != Fonts.end();) {
-		if ((font->first & 0xFFFF0000) == fontId) {
+		if ((font->first & 0xFFFF0000) == fontStyle) {
 			Fonts.erase(font++);
 		} else {
 			font++;

```

Comment:
I don't disagree with this suggestion, but just because I found it interesting.

> https://en.cppreference.com/w/cpp/container/unordered_map/erase
> References and iterators to the erased elements are invalidated. Other iterators and references are not invalidated.